### PR TITLE
Short-circuit functions with zero dim results for IE backend

### PR DESCRIPTION
**Pull request by adk9**
OV Inference Engine backends do not support zero dim tensors or results. In this PR, we mark such zero-dim result functions as trivial functions and short-circuit the whole cluster by turning into a NOP. This works because all clusters encapsulated by the bridge, at present, are strictly stateless.

### CODE REVIEW

**Review by bani-intelaipg — APPROVED** [review]
Some tests are failing, which need investigation. Code changes look good.

File: test/python/tensorflow/tests_linux_ie_cpu.txt [hunk 1]
```diff
@@ -44,6 +44,10 @@ argmax_op_test.ArgMaxTest.testOutputEmpty
 argmax_op_test.ArgMaxTest.testEmpty
 array_ops_test.SnapshotOpTest.testInvertPermutation
 array_ops_test.StridedSliceGradTest.testInt64Indices
+array_ops_test.StridedSliceTest.testExpand
```
Comment (bani-intelaipg) [inline test/python/tensorflow/tests_linux_ie_cpu.txt:47]: Can we put some comments as to why these are skipped?

**Review by bani-intelaipg — CHANGES_REQUESTED** [review]

File: test/python/tensorflow/tests_linux_ie_cpu.txt (hunk 1 above)
Comment (kanvi-nervana) [reply test/python/tensorflow/tests_linux_ie_cpu.txt:47]: I think these are enabled now with the change

File: ngraph_bridge/ie_executable.cc [hunk 2]
```diff
@@ -40,12 +40,18 @@ IE_Executable::IE_Executable(shared_ptr<Function> func, string device)
...
+  //  3. zero function (* -> Zero)
   NGRAPH_VLOG(2) << "Checking for trivial functions in IE backend";
   bool trivial_fn = true;
   for (auto result : func->get_results()) {
     auto parent = result->input_value(0).get_node_shared_ptr();
+    auto& shape = result->get_shape();
     trivial_fn &= ngraph::is_type<opset::Parameter>(parent) ||
-                  ngraph::is_type<opset::Constant>(parent);
+                  ngraph::is_type<opset::Constant>(parent) ||
+                  count(shape.begin(), shape.end(), 0);
```
Comment (kanvi-nervana) [inline ngraph_bridge/ie_executable.cc:54]: Why do we have the zero dim check at 2 places?

File: test/python/tensorflow/tests_linux_ie_cpu.txt (hunk 1 above)
Comment (adk9) [reply test/python/tensorflow/tests_linux_ie_cpu.txt:47]: That's correct.

File: ngraph_bridge/ie_executable.cc (hunk 2 above)
Comment (adk9) [reply ngraph_bridge/ie_executable.cc:54]: This instance marks the function as a trivial function. When the trivial function is being executed, we handle all three cases separately and have to check what type of trivial function it is.

**Review by kanvi-nervana — APPROVED** [review]

**Pull request merged**